### DISCUSSION TITLE
sending a message to a specific user directly

### DISCUSSION THREAD

**Discussion by samialtaher** [cmgmyr/laravel-messenger#388] [General]
First please know that i am not a developer, but can follow code and get by :)
I am using this package on my new site. However, I have groups of users, so the user will only see other users in their group to send messages to. This is working fine.
I also have a  page listing all users in the network, and want a way to allow people to click a button (send message) to send a message to any user (even if not in their group). 
Can someone please show me / guide me with an example - I am using Laravel 8 

I tried something like a route/link  like this: site.com/messages/create?recipient=ID    -  where ID is the user ID of the person to send a message to. But no variation would work.

Thanks

**Comment by cmgmyr** [comment]
You can handle this a few different ways. In the examples [here](https://github.com/cmgmyr/laravel-messenger/blob/master/examples/views/messenger/create.blade.php#L24) we list out any possible users/recipients to join a thread, and those get processed [here](https://github.com/cmgmyr/laravel-messenger/blob/master/examples/MessagesController.php#L105)

If you want a user to send a message to everyone in a group you could swap [this](https://github.com/cmgmyr/laravel-messenger/blob/master/examples/MessagesController.php#L105) out for logic to gather the user ids then pass it into `$thread->addParticipant($userIds)`.

I hope that helps
